types/result.go

```diff
@@ -8,10 +8,12 @@ import (
 
 // Result is a common result object for ABCI calls.
 // CONTRACT: a zero Result is OK.
+// DEPRECATED: prefer raw types instead.
```

Comment:
++.

But we're making a breaking change. Let's just remove the file? I can't imagine it will be useful to anyone if all the interfaces are abandoning it :)